doc/api/esm.md

```diff
@@ -156,15 +159,6 @@ via the paths defined in [`"exports"`][].
 For details on these package resolution rules that apply to bare specifiers in
 the Node.js module resolution, see the [packages documentation](packages.md).
 
-### Mandatory file extensions
-
-A file extension must be provided when using the `import` keyword to resolve
-relative or absolute specifiers. Directory indexes (e.g. `'./startup/index.js'`)
-must also be fully specified.
-
-This behavior matches how `import` behaves in browser environments, assuming a
-typically configured server.
```

Comment:
This is referring to `import`, not entry points, and shouldn’t be removed. I think the docs for this PR would more or less match the reverse of the changes in https://github.com/nodejs/node/pull/31415.